File: App_architecte_back/services/notification_service.py
import requests
from models import Notification, User
from extensions import db

EXPO_URL = "https://exp.host/--/api/v2/push/send"


def send_notification(token, title, message, user_id=None):

    # 🔔 Payload Expo
    payload = {
        "to": token,
        "sound": "default",
        "title": title,
        "body": message,
    }

    try:
        response = requests.post(EXPO_URL, json=payload)

        if response.status_code != 200:
            print("Expo error:", response.text)

    except Exception as e:
        print("Erreur notification:", e)

    # 💾 Save notification in DB
    if user_id:
        notif = Notification(
            user_id=user_id,
            title=title,
            message=message
        )
        db.session.add(notif)
        db.session.commit()


# 🔹 Helper pour récupérer le token utilisateur
def get_user_token(user_id):
    user = User.query.get(user_id)
    return user.push_token if user else None

File: appArchitecte_front/src/services/notificationService.js
import api from "./api";

// notificationService.js
const notificationService = {
  // ✅ Enregistre le token Expo auprès du backend
  saveToken: async (token) => {
    try {
      // ⚠️ Endpoint corrigé : /api/save-token
      const res = await api.post("/api/save-token", { token });
      return res.data;
    } catch (err) {
      console.log("Erreur save token:", err.response?.data || err.message);
      throw err;
    }
  },

  // ✅ Récupère les notifications pour l'utilisateur connecté
  getNotifications: async () => {
    try {
      const res = await api.get("/api/notifications");
      return res.data;
    } catch (err) {
      console.log("Erreur get notifications:", err.response?.data || err.message);
      return [];
    }
  },
};

export default notificationService;